### PR TITLE
(2723) ISPF themes support multiple selection

### PR DESCRIPTION
**Pull request by CristinaRO**
## Changes in this PR
- ISPF activities can have multiple ISPF themes

### Before
Screenshot 2022-12-06 at 21 59 19
Screenshot 2022-12-06 at 21 59 13

### After
Screenshot 2022-12-06 at 21 49 33
Screenshot 2022-12-06 at 21 49 24

## Next steps

- [ ] Is an ADR required? An ADR should be added if this PR introduces a change to the architecture.
- [x] Is a changelog entry required? An entry should always be made in `CHANGELOG.md`, unless this PR is a small tweak which has no impact outside the development team.
- [ ] Do any environment variables need amending or adding?
- [ ] Have any changes to the XML been checked with the IATI validator? See [XML Validation](https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/blob/develop/doc/xml-validation.md)

### CODE REVIEW

**Comment by CristinaRO** [comment]
[Make ISPF themes multiple selection (checkboxes rather than radio buttons)](https://trello.com/c/TOknUp5f/2723-make-ispf-themes-multiple-selection-checkboxes-rather-than-radio-buttons)

File: db/migrate/20221206171338_change_ispf_theme_to_ispf_themes_array.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class ChangeIspfThemeToIspfThemesArray < ActiveRecord::Migration[6.1]
+  def change
+    rename_column :activities, :ispf_theme, :ispf_themes
+    change_column :activities, :ispf_themes, :integer, array: true, using: "(array[ispf_themes]::integer[])"
```
Comment (Gweaton) [inline db/migrate/20221206171338_change_ispf_theme_to_ispf_themes_array.rb:4]: I've never had to do this before in Rails but I'm struggling to find documentation for how to use `using` here - did you find any?

This is more for my own understanding than anything else!

**Review by Gweaton — APPROVED** [review]
This looks good to me, I've tested locally to make sure existing activities with a single ISPF theme don't break after this change (which I thought we might have to handle manually but it looks like ActiveRecord does a lot of this for us?)

File: db/migrate/20221206171338_change_ispf_theme_to_ispf_themes_array.rb (hunk 1 above)
Comment (CristinaRO) [reply db/migrate/20221206171338_change_ispf_theme_to_ispf_themes_array.rb:4]: This is where I found how to convert the data type while keeping the value: https://stackoverflow.com/questions/13879713/postgresql-alter-column-data-type-from-integer-to-integer-array

**Pull request merged**